### PR TITLE
New graphing modes for --graph

### PR DESCRIPTION
**Pull request by Camerooooon**
I added multiple new modes for different types of graphing. This includes graphing the CPU usage, CPU temperature, and CPU frequency. I also changed the way we parse the --graph command so that the user can provide a graph type.

### CODE REVIEW

**Comment by Camerooooon** [comment]
This is a massive PR please make sure to test rigorously and leave comments!

**Comment by Camerooooon** [comment]
Implements #285

**Comment by JakeRoggenbuck** [comment]
Great job, this looks really good, I think I have one or two things of feedback.

File: src/main.rs [hunk 1]
```diff
@@ -256,10 +258,20 @@ fn parse_args(config: config::Config) {
         } => {
             if !config_dir_exists() {
                 warn_user!("Config directory '/etc/acs' does not exist!");
+                thread::sleep(time::Duration::from_millis(5000));
             }
 
+            let parsed_graph_type = match get_graph_type(&should_graph) {
+                Some(graph_type) => graph_type,
+                None => {
+                    warn_user!("Graph type is not set! Can be freq, usage, or temp Continuing in 5 seconds...");
```
Comment (JakeRoggenbuck) [inline src/main.rs:267]: I think we should just let GraphType be hidden without telling the user that there is a warning

Comment (JakeRoggenbuck) [reply src/main.rs:267]: Would it be possible to show the warning if the user does the -g flag without an arg to help the user out, but not if the don't include the flag?

**Comment by JakeRoggenbuck** [comment]
Everything graphing works as intended and a very good solution!

**Comment by JakeRoggenbuck** [comment]
Laptop tested

File: src/main.rs (hunk 1 above)
Comment (Camerooooon) [reply src/main.rs:267]: Ah yes, I see what you mean, the intended behavior was for the warning to show only if the graph arg is stated but the graph type is invalid or not set. Yes, I'll fix that thank you.

Comment (JakeRoggenbuck) [reply src/main.rs:267]: Thanks!

**Review by Shuzhengz — APPROVED** [review]
ayy

**Review by JakeRoggenbuck — APPROVED** [review]

**Review by Shuzhengz — APPROVED** [review]

**Pull request merged**